src/workerd/jsg/resource.h

```diff
@@ -966,6 +1086,28 @@ struct ResourceTypeBuilder {
             v8::Local<v8::Value>(), signature, 0, v8::ConstructorBehavior::kThrow));
   }
 
+  // Register a method with both slow and fast paths using the V8 Fast API mechanism
+  template <const char* name, typename Method, Method method, bool trackCallCount = false>
+  inline void registerFastMethod() {
+    static_assert(IsFastMethodCompatible<Method>::value);
+    // Create the C function for the fast path
+    auto cFunction = FastMethodCallback<TypeWrapper, name, isContext, Self, Method, method,
+        ArgumentIndexes<Method>>::template makeCFunction<trackCallCount>();
+
+    // Create a v8::MemorySpan to hold the CFunction
+    // TODO(soon): Add support for multiple CFunctions.
+    v8::MemorySpan<const v8::CFunction> cfunctionSpan(&cFunction, 1);
+
+    // Create a function template with both slow and fast paths
+    auto tmpl = v8::FunctionTemplate::NewWithCFunctionOverloads(isolate,
+        &FastMethodCallback<TypeWrapper, name, isContext, Self, Method, method,
+            ArgumentIndexes<Method>>::template slowCallback<trackCallCount>,
+        v8::Local<v8::Value>(), signature, 0, v8::ConstructorBehavior::kThrow,
+        v8::SideEffectType::kHasNoSideEffect, cfunctionSpan);
```

Comment:
hmm.. is `kHasNoSideEffect` correct? I'm not sure we can necessarily guarantee that a JSG_METHOD has no side effect.

Reply:
I updated the implementation. (used kHasSideEffect)